web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/behandling/arbeidInntektsmelding/ArbeidOgInntektsmeldingDtoTjeneste.java

```diff
@@ -148,33 +148,26 @@ private List<InntektsmeldingDto> mapInntektsmeldinger(List<Inntektsmelding> innt
             .filter(Behandling::erYtelseBehandling)
             .toList();
 
-        var mapAvInntektsmeldingerTilBehandlingsIder = behandlinger.stream()
-            .flatMap(behandling -> inntektsmeldingTjeneste.hentAlleInntektsmeldingerForAngitteBehandlinger(Set.of(behandling.getId()))
-                .stream()
-                .map(im -> Map.entry(im.getIndexKey(), behandling.getUuid())))
-            .collect(Collectors.groupingBy(Map.Entry::getKey, Collectors.mapping(Map.Entry::getValue, Collectors.toList())));
-
         var arbeidsforholdValgListe = arbeidsforholdInntektsmeldingMangelTjeneste.hentArbeidsforholdValgForSak(referanse);
 
         var mapAvAktiveInntektsmeldingerTilBehandlingsIder = behandlinger.stream()
-            .flatMap(behandling -> inntektsmeldingTjeneste.hentInntektsmeldinger(referanse, referanse.getUtledetSkjæringstidspunkt())
+            .flatMap(behandling -> inntektsmeldingTjeneste.hentInntektsmeldinger( BehandlingReferanse.fra(behandling), referanse.getUtledetSkjæringstidspunkt())
                 .stream()
                 .filter(im -> arbeidsforholdValgListe.stream().noneMatch(arbeidsforholdValg -> {
                     var matchendeArbeidsforhold = arbeidsforholdValg.getArbeidsgiver().getIdentifikator().equals(im.getArbeidsgiver().getIdentifikator());
-                    var harValgtFortsetteUtenIM = arbeidsforholdValg.getVurdering() == ArbeidsforholdKomplettVurderingType.FORTSETT_UTEN_INNTEKTSMELDING;
+                    var harValgtFortsetteUtenIM = arbeidsforholdValg.getVurdering() == ArbeidsforholdKomplettVurderingType.IKKE_OPPRETT_BASERT_PÅ_INNTEKTSMELDING;
                     return matchendeArbeidsforhold && harValgtFortsetteUtenIM;
                 }))
-                .map(im -> Map.entry(im.getIndexKey(), behandling.getUuid())))
+                .map(im -> Map.entry(im.getJournalpostId(), behandling.getUuid())))
```

Comment:
indexKey ikke unikt når IM er sendt inn for samme arbeidsgiver. Derfor journalpostId